core/src/ldml/ldml_transforms.cpp

```diff
@@ -831,6 +844,67 @@ transforms::load(
   return transforms;
 }
 
+// string
+
+// TODO-LDML: copypasta -> refactor
+std::u32string &normalize_nfd(std::u32string &str, UErrorCode &status) {
+  std::u16string rstr = km::kbp::kmx::u32string_to_u16string(str);
+
+  normalize_nfd(rstr, status);
+  if (U_SUCCESS(status)) {
+    // if failure, leave it alone
+    str = km::kbp::kmx::u16string_to_u32string(rstr);
+  }
+  return str;
```

Comment:
I think I'd prefer the failure cases in each function to be tested and `return str` just like the following functions do, for consistency.

Also, is there a good reason to return str as well as modify the input parameter? This seems likely to lead to unintentional wrong uses of the functions. Why not return bool success value instead (as suggested earlier in my review)?